test/unit/model/ProctorioApiServiceTest.php

```diff
@@ -0,0 +1,134 @@
+
+        $this->deliveryExecution = $this->getDeliveryExecution();
+        $this->subject = new ProctorioApiService();
+
+        /** @var ProctorioService|MockObject $proctorioLibraryMock */
+        $proctorioLibraryMock = $this->getMockBuilder(ProctorioService::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $proctorioLibraryMock->method('callRemoteProctoring')
+            ->willReturn('["ttURL","reviewURL"]');
+
+        $this->subject->setProctorioUrlLibraryService($proctorioLibraryMock);
+        $this->subject->setServiceLocator($serviceLocatorMock);
+        $this->setLogger();
+    }
+
+    public function testGetProctorioUrl(): void
+    {
+        $this->subject->setOption(ProctorioApiService::OPTION_OAUTH_KEY, 'key');
+        $this->subject->setOption(ProctorioApiService::OPTION_OAUTH_SECRET, 'secret');
+        $expected = new ProctorioResponse('ttURL', 'reviewURL');
+        $proctorioResponse = $this->subject->getProctorioUrl($this->deliveryExecution);
+        $this->assertInstanceOf(ProctorioResponse::class, $proctorioResponse);
+        $this->assertEquals($expected, $proctorioResponse);
+    }
+
+    private function getDeliveryExecution()
+    {
+        $this->deliveryExecution = $this->getMockBuilder(DeliveryExecutionInterface::class)
+            ->getMock();
+
+        $delivery = $this->getMockBuilder(core_kernel_classes_Resource::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $delivery->method('getLabel')->willReturn('test-Label');
+
+        $this->deliveryExecution->method('getUserIdentifier')
+            ->willReturn('test');
+        $this->deliveryExecution->method('getDelivery')
+            ->willReturn($delivery);
+
+        $this->deliveryExecution->method('getIdentifier')
+            ->willReturn('ttURL');
+
+        return $this->deliveryExecution;
+    }
+
+    protected function getPersistance(): MockObject
+    {
+        $persistance = $this->getMockBuilder(common_persistence_KeyValuePersistence::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $persistanceManager = $this->getMockBuilder(PersistenceManager::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $persistanceManager->method('getPersistenceById')->willReturn($persistance);
+        return $persistanceManager;
+    }
+
+
+    protected function setMockServices(MockObject $persistanceManager): ServiceLocatorInterface
+    {
+        $services = [
+            PersistenceManager::SERVICE_ID => $persistanceManager,
+            LaunchService::class => $this->getMockBuilder(LaunchService::class)->getMock(),
+            ProctorioRequestBuilder::class => $this->getMockBuilder(ProctorioRequestBuilder::class)->getMock(),
+        ];
+
+        return $this->getServiceLocatorMock($services);
+    }
+
+    protected function setLogger(): voidgetServiceLocatorMock
```

Comment:
That is typo, correct? 
```suggestion
    protected function setLogger(): void
```